modules/javafx.graphics/src/main/java/javafx/scene/image/PixelBuffer.java

```diff
@@ -0,0 +1,194 @@
+     * @throws IllegalArgumentException if either {@code pixelBuffer} dimension
+     *                                  is negative or zero or if {@code pixelFormat}
+     *                                  is unsupported or if the  {@code buffer} does
+     *                                  not have sufficient memory.
+     * @throws NullPointerException     if {@code buffer} is {@code null}.
+     */
+    public PixelBuffer(int width, int height, T buffer, PixelFormat pixelFormat) {
+        Objects.requireNonNull(buffer, "A valid buffer must be provided.");
+        if (width <= 0 || height <= 0) {
+            throw new IllegalArgumentException("PixelBuffer dimensions must be positive (w,h > 0)");
+        }
+        if (pixelFormat.getType() != PixelFormat.getByteBgraPreInstance().getType() &&
+                pixelFormat.getType() != PixelFormat.getIntArgbPreInstance().getType()) {
+            throw new IllegalArgumentException("Unsupported PixelFormat: " + pixelFormat.getType());
+        }
+        if ((pixelFormat.getType() == PixelFormat.getByteBgraPreInstance().getType()
+                && buffer.capacity() < width * height * 4) ||
+                (pixelFormat.getType() == PixelFormat.getIntArgbPreInstance().getType()
+                && buffer.capacity() < width * height)) {
+            throw new IllegalArgumentException("Insufficient memory allocated for buffer.");
+        }
+        this.buffer = buffer;
+        this.width = width;
+        this.height = height;
+        this.pixelFormat = pixelFormat;
+    }
+
+    /**
+     * Returns the {@code java.nio.Buffer} of this {@code PixelBuffer}.
+     *
+     * @return the {@code java.nio.Buffer} of this {@code PixelBuffer}.
+     */
+    public T getBuffer() {
+        return buffer;
+    }
+
+    /**
+     * Returns the width dimension of {@code PixelBuffer}.
+     *
+     * @return the width dimension of this {@code PixelBuffer}.
+     */
+    public int getWidth() {
+        return width;
+    }
+
+    /**
+     * Returns the height dimension of {@code PixelBuffer}.
+     *
+     * @return the height dimension of this {@code PixelBuffer}.
+     */
+    public int getHeight() {
+        return height;
+    }
+
+    /**
+     * Returns the {@code PixelFormat} of the pixels in the {@code java.nio.Buffer}.
+     *
+     * @return the {@code PixelFormat} of the pixels in the {@code java.nio.Buffer}.
+     */
+    public PixelFormat getPixelFormat() {
+        return pixelFormat;
+    }
+
+    /**
+     * Updates the pixel data in {@code java.nio.Buffer} using provided {@link javafx.util.Callback}.
+     * This method must be called on the JavaFX Application thread.
+     * The {@code callback} method should update the {@code java.nio.Buffer} and
+     * return a {@code Rectangle2D} which encloses the modified region, or
+     * {@code null} to indicate that entire buffer is dirty.
+     * Pseudo-code how to use this method:
```

Comment:
I recommend adding a `<p>` paragraph tag before this. Also, something more like this:

```
    * <p>Example code that shows how to use this method:
```